src/chain-cache/ChainSync.ts

```diff
@@ -268,4 +286,62 @@ export class ChainSync {
     };
     setTimeout(processEvents, 1);
   }
+  private _resetPairsFetching() {
+    this._pairs = [];
+    this._slowPollPairs = false;
+    this._initialSyncDone = false;
+  }
+
+  private async _detectReorg(currentBlock: number): Promise<boolean> {
+    logger.debug('_detectReorg called');
+    const blocksMetadata: BlockMetadata[] = this._chainCache.blocksMetadata;
+    const numberToBlockMetadata: { [key: number]: BlockMetadata } = {};
+    for (let blockMetadata of blocksMetadata) {
+      const { number, hash } = blockMetadata;
+      if (number > currentBlock) {
+        console.log(
```

Comment:
logger.debug?

Reply:
changed to logger and added control over verbosity